File: template.py
#!/usr/bin/env python3

"""
Nom du script: script.py
Description: [Brève description de ce que fait le script]
Auteur: [Votre nom]
Date de création: [Date de création]
Version: [Version du script]
Licence: [Licence du script]
"""

import argparse

def parse_arguments():
    """Parse les arguments de la ligne de commande."""
    parser = argparse.ArgumentParser(description='[Description détaillée du script]')
    parser.add_argument('--param1', required=True, type=str, help='Description du premier paramètre')
    parser.add_argument('--param2', required=False, type=int, default=0, help='Description du deuxième paramètre')
    parser.add_argument('--version', action='version', version='%(prog)s 1.0')
    return parser.parse_args()

def main():
    """Point d'entrée principal du script."""
    args = parse_arguments()

    # Validation des paramètres
    if args.param1 == '':
        print("Le paramètre --param1 est obligatoire.")
        exit(1)

    if args.param2 < 0:
        print("Le paramètre --param2 doit être un nombre positif.")
        exit(1)

    # Logique principale du script
    print(f"Paramètre 1: {args.param1}")
    print(f"Paramètre 2: {args.param2}")

    # Ajoutez ici la logique principale de votre script
    example_function(args.param1, args.param2)

def example_function(param1, param2):
    """Une fonction d'exemple."""
    print(f"Exécution de example_function avec les paramètres: {param1}, {param2}")
    # Ajoutez ici le code de la fonction

if __name__ == "__main__":
    main()

File: template.php
<?php
/**
 * Nom du script : script.php
 * Description   : [Brève description de ce que fait le script]
 * Auteur        : [Votre nom]
 * Date de création : [Date de création]
 * Version       : [Version du script]
 * Licence       : [Licence du script]
 */

// Inclusion des fichiers nécessaires
require 'config.php';
// require 'autre_fichier.php'; // Décommentez si nécessaire

/**
 * Fonction pour traiter les paramètres
 *
 * @param array $params Paramètres d'entrée
 * @return void
 */
function processParams($params) {
    if (empty($params['param1'])) {
        echo "Le paramètre 'param1' est obligatoire.\n";
        exit(1);
    }

    $param1 = $params['param1'];
    $param2 = isset($params['param2']) ? $params['param2'] : 0;

    // Validation des paramètres
    if ($param2 < 0) {
        echo "Le paramètre 'param2' doit être un nombre positif.\n";
        exit(1);
    }

    // Appel de la fonction principale
    exampleFunction($param1, $param2);
}

/**
 * Fonction principale de l'exemple
 *
 * @param string $param1 Premier paramètre
 * @param int $param2 Deuxième paramètre
 * @return void
 */
function exampleFunction($param1, $param2) {
    echo "Exécution de exampleFunction avec les paramètres : $param1, $param2\n";
    // Ajoutez ici le code de la fonction
}

// Récupération des paramètres depuis la ligne de commande
$params = getopt('', ['param1:', 'param2::']);

// Appel de la fonction pour traiter les paramètres
processParams($params);
?>
